nservicebus/upgrades/5to6-async-suffix.md

```diff
@@ -0,0 +1,116 @@
+ * Classes are not suffixed with "Instance" or "Static".
+ * Members are not suffixed [Access modifier names](https://msdn.microsoft.com/en-au/library/ms173121.aspx) such as "Protected" or "Public".
+
+All these things can be inferred by the IDE and the compiler and appropriate IntelliSense and compiler messages are provided to the developer.
+
+So in deciding on the adoption of the *Async* suffix it was necessary to choose between consistency with certain external .NET APIs or naming consistence within NServiceBus.
+
+See also: [Hungarian notation Disadvantages](https://en.wikipedia.org/wiki/Hungarian_notation#Disadvantages)
+
+
+### Async APIs should be identifiable in code
+
+One of the arguments for the *Async* suffix is that all async should be clearly identifiable in code so as to prevent miss-use of that API. However the Compiler is very efficient at identifying incorrect async member usage and providing appropriate feedback to the developer. Some possible miss-uses are listed below with the associated compiler information.
+
+Given an async API being used:
+
+snippet: ServiceWithAsync
+
+
+#### Missing await
+
+When a void method calls an async method but neglects to await that method.
+
+snippet: VoidMethodMissingAwait
+
+Results in [Compiler Warning CS4014](https://msdn.microsoft.com/en-us/library/hh873131.aspx)
+
+
+#### Missing return task
+
+When a Task method calls an async method but neglects to await that method.
+
+snippet: TaskMethodMissingAwait
+
+Results in [Compiler Error CS0161](https://msdn.microsoft.com/en-us/library/87cz4k9t.aspx)
+
+
+#### Async method with missing await
+
+When a async Task method calls an async method but neglects to await that method.
+
+snippet: AsyncMethodMissingAwait
+
+Results in [Compiler Warning CS4014](https://msdn.microsoft.com/en-us/library/hh873131.aspx)
+
+
+#### Missing a single await
+
+When a async Task method awaits one async method but neglects to await another.
+
+snippet: AsyncMethodMissingOneAwait
+
+Results in [Compiler Warning CS4014](https://msdn.microsoft.com/en-us/library/hh873131.aspx)
+
+
+#### Treat Warnings as errors
+
+Note that in several of the above examples are warnings and not errors. As such it is necessary to either [Treat all Warnings as Errors](https://msdn.microsoft.com/en-us/library/kb4wyys2.aspx#Anchor_3) or nominate specific warnings to be treated as errors via [Errors and Warnings](https://msdn.microsoft.com/en-us/library/kb4wyys2.aspx#Anchor_2).
+
+
+### Async not necessary when reading code
+
+The above examples show how it is difficult to incorrectly use async APIs appropriate. As such async API usage is clearly identifiable in code by the associated `await`, `.ConfigureAwait()` usage that is required.
+
+
+## Other libraries with no Async suffix.
+
+Other libraries are also taking the same approach. For example:
+
+ * [Octokit - The GitHub .net API](https://github.com/octokit/octokit.net)
+ * [MassTransit](http://masstransit-project.com/)
```

Comment:
Quartz.NET can be added as well see https://github.com/quartznet/quartznet/issues/331